src/recon_buildblock/PoissonLogLikelihoodWithLinearModelForMeanAndListModeDataWithProjMatrixByBin.cxx

```diff
@@ -325,6 +341,194 @@ warning("PoissonLogLikelihoodWithLinearModelForMeanAndListModeDataWithProjMatrix
 return true;
     }
 
+  if(cache_size > 0)
+      cache_lm_file = true;
+
+  if(!recompute_cache && cache_lm_file)
+  {
+      std::string curr_dir = FilePath::get_current_working_directory();
+      std::string cache_filename = "my_CACHE00.bin";
```

Comment:
Should this filename be hardcoded?

Reply:
At this point, I would prefer it hardcoded for safety. In the scenario that the RAM is not enough to accommodate the full listmode file, or in any case that ```cache_size < number_of_events```, there is no mechanism to have multiple passes or multiple cache files. Therefore it might be safer to leave it hardcoded. 
I should note that I haven't run in such case, yet.

Reply:
In my private code, I use a function get_total_number_of_events() from the InputStream class. But not all InputStreams support this. So, I can really bring that here without discussion.